openquake/hazardlib/source/non_parametric.py

```diff
@@ -66,6 +66,16 @@ def __init__(self, source_id, name, tectonic_region_type, data,
             for (rup, pmf), weight in zip(data, weights):
                 rup.weight = weight
 
+    @property
+    def num_ruptures(self):
+        if not hasattr(self, '_num_ruptures'):
+            self._num_ruptures = len(self.data)
+        return self._num_ruptures
+
+    @num_ruptures.setter
```

Comment:
I do not understand this. Setting `num_ruptures` is potentially dangerous as it could break the relation with `len(self.data)`. Why do we need this?